src/renku_crac/app.py

```diff
@@ -0,0 +1,369 @@
+                    for r in res
+                ]
+            )
+
+        return handler
+
+    def _get_resource_pool_user(self):
+        async def handler(_: Request, resource_pool_id: int, user_id: str):
+            res = await self.config.db.get_users(keycloak_id=user_id, resource_pool_id=resource_pool_id)
+            if len(res) < 1:
+                raise errors.MissingResourceError(
+                    message=f"The user with id {user_id} or resource pool with id {resource_pool_id} cannot be found."
+                )
+            return json(
+                apispec.UserWithId(id=res[0].keycloak_id, username=cast(EmailStr, res[0].username)).dict(
+                    exclude_none=True
+                )
+            )
+
+        return handler
+
+    def _get_resource_pool_classes(self):
+        async def handler(request: Request, resource_pool_id: int):
+            res = await self.config.db.get_classes(resource_pool_id=resource_pool_id, name=request.args.get("name"))
+            return json([apispec.ResourceClassWithId.from_orm(r).dict(exclude_none=True) for r in res])
+
+        return handler
+
+    def _get_resource_pool_class(self):
+        async def handler(_: Request, resource_pool_id: int, class_id: int):
+            res = await self.config.db.get_classes(resource_pool_id=resource_pool_id, id=class_id)
+            if len(res) < 1:
+                raise errors.MissingResourceError(
+                    message=f"The class with id {class_id} or resource pool with id {resource_pool_id} cannot be found."
+                )
+            return json(apispec.ResourceClassWithId.from_orm(res[0]).dict(exclude_none=True))
+
+        return handler
+
+    def _delete_resource_pool_class(self):
+        async def handler(_: Request, resource_pool_id: int, class_id: int):
+            await self.config.db.delete_resource_class(resource_pool_id=resource_pool_id, resource_class_id=class_id)
+            return HTTPResponse(status=204)
+
+        return handler
+
+    def _put_patch_resource_pool_class(self, validation: Type[apispec.ResourceClassPatch | apispec.ResourceClass]):
+        @validate(json=validation)
+        async def handler(
+            _: Request, resource_pool_id: int, class_id: int, body: apispec.ResourceClassPatch | apispec.ResourceClass
+        ):
+            cls = await self.config.db.update_resource_class(
+                resource_pool_id=resource_pool_id, resource_class_id=class_id, **body.dict(exclude_none=True)
+            )
+            return json(apispec.ResourceClassWithId.from_orm(cls).dict(exclude_none=True))
+
+        return handler
+
+    def _post_resource_pool_class(self):
+        @validate(json=apispec.ResourceClass)
+        async def handler(_: Request, body: apispec.ResourceClass, resource_pool_id: int):
+            rps = await self.config.db.get_resource_pools(id=resource_pool_id)
```

Comment:
Are duplicate names for resource classes an issue? If so we should check for it here and add a unique constraint on the DB

Reply:
I do not want to enforce unique names on resource classes at all. So if users want to make several resource classes with the same name they can do so. We will always go by the unique id in the db. The names are always for display purposes only.